src/views/sidebar/utils.ts

```diff
@@ -248,3 +249,47 @@ export const getFolderTranslatedName = ({ folderId, folderName }: GetSystemFolde
 
 	return folderName;
 };
+
+/**
+ * Returns the root account name for a given folder
+ * @param folder a Folder or LinkFolder
+ * @returns the root account name or null if the folder is not a link or the root folder
+ */
+export const getRootAccountName = (folder: Folder | LinkFolder): string | null => {
```

Comment:
About those functions:
IMO they should be moved inside the _helpers/folders.ts_ file which should contains all the folders-related functions and utilities.
The getParentId (whose name is misleading) is redundant since there is a function getFolderIdParts (also in the _helpers/folders.ts_) which return the desired id.

Anyway this is my bad: I didn't share the information about those helper functions. I will document them. Sorry

Reply:
I have moved the functions to the helpers folder, but I think we should discuss where to position utils function as the whole project is currently positioning utils functions in folders located close to the code that they serve.
I think your idea to position functions in a more general location is preferable, we should just discuss it a team level and implement it retrospectively. probably we will spot out more duplications.

I have renamed getParentId to getSystemFolderParentId, which is more explicative.
regarding getFolderIdParts is however not working, as it always returning the same result: const result is assigned at the first line, and subsequently returned as default or by different if statements.
it will always return { zid: null, id: null }

Reply:
I didn't mean that was mandatory to move the functions. In my mind that was supposed to be just a side note for future implementation, but I wasn't clear. As I wrote, during the PR I realized that I missed to discuss with you about the helpers functions and I didn't share info about my implementation. For this reason I had already put the topic inside the retrospective Miro board.

About the _getFolderIdParts_: the function is working properly (also covered by unit tests). The null values are returned only when the parameter is missing or when the parameter doesn't match the pattern. The single parts are set at the line 37 where it is destructoring the result from the regex parse result